.prettierrc

```diff
@@ -0,0 +1,12 @@
+useTabs: true
+tabWidth: 2
+printWidth: 100
+singleQuote: true
+trailingComma: es5
+bracketSpacing: true
+parenSpacing: true
+jsxBracketSameLine: false
+insertPragma: true
+requirePragma: false
```

Comment:
Am I reading it right that we'll format files independently of `@format` pragma?

We'll likely need to add a `.prettierignore` and modify this script to follow the same rules or make other adjustments to ensure prettier behaves consistently across editor plugins and scripts (pre-commit/reformat-files).

https://github.com/Automattic/wp-calypso/blob/d6cc27208336adf23cd8e33dee3f07d1e3252d20/bin/utils/should-format.js

Some related discussion (p4TIVU-8Ud-p2 #comment-8921)

cc: @blowery @samouri

Reply:
> Am I reading it right that we'll format files independently of @format pragma?

Setting `requirePragma` to `false` means that if I tell Prettier to format the file (from command line, from editor plugin, from a script), it will format it even if the pragma is not present. If the option was `true`, Prettier would never format and would always return the original text for files without the pragma.

The precommit hooks and other scripts can override the option explicitly if they want to format only files that have the pragma. But when hitting the Ctrl-Alt-F shortcut in my Atom, it's expected that the file is formatted no matter what, isn't it?

> If it's not required, should we also not insert it? Can we remove it?

The `pre-commit-hook.js` and `reformat-files.js` scripts implement their own pragma check, so it's not entirely true that it's not required. That code dates back to the time when Prettier didn't have the pragma support is has today. @samouri himself contributed a big part of this functionality to the official project.